Crop, Livestock, and Property Resource Implementations.

nickwxo opened the pull request:
This pull request implements the Crop, Livestock, and Property endpoints as well as the ACL files for them. They are all properly authenticated by token.

Commit:
Crop, Livestock, and Property Resource Implementations.

Commit:
Important correction. We should have been pulling all these from the database views Wayne created - not the tables themselves.

matjames007 commented on routes/resources/crop/crop.js:
```diff
@@ -0,0 +1,58 @@
+};
+
+
+exports.getCropByID = function(req, res, next) {
```
@nick22891 implementation needed.

matjames007 commented on routes/resources/property/router-property.js:
```diff
@@ -0,0 +1,15 @@
+ * End Points relevant to Farmers
+ */
+router.get('/properties', Property.getAllProperties);
+router.get('/property/:id', Property.getPropertyByID);
```
@nick22891 implement all the `getXByID` functions.

matjames007 reviewed (dismissed):
Still need to implement a few of more of the ACL + resources endpoints.

Commit:
Implemented selection of a specific farmer, crop, property, or livestock by their respective IDX number. Also switched from using the views to the new tables Hartnell and I created.

nickwxo commented:
@matjames007 This pull request can be reviewed again.

Commit:
Sequelize library working in that it returns all the requested data correctly but it also raises an error in the JSON that is of some concern and affects ACL filtering.

Commit:
The farmers resource can now be accessed properly with no errors using the Sequelize library. More importantly, we can now specify a limit and offset parameter to effectively paginate the data and we can also search by specific fields.

Commit:
The Sequelize library has been implemented for the remaining (Property, Crop, and Livestock) endpoints along with pagination and some search functionality as explained earlier.

Commit:
Merge branch 'develop' into harvest-additional-endpoints-impl

* develop: (70 commits)
  This commit should rectify the issue pointed out in #72 in the issue tracking tool.
  Needed in order to force an angular dependency of 1.5.8
  Adding missing Angular table dependency.
  Added CSS for Developer Dashboard UI, added 'Cancel' button to "Add a new App" modal and fixed "Change Password" modal.
  This commit should rectify issue #53 as the user object is now shared with navigation controller allowing updates throughout the screens (if necessary)
  This commit serves as a demonstration of: * commenting, which as previously mentioned was lacking throughout the corresponding pull request * code reuse, utililzing LogsFactory throughout the angular application without creating more footprint * reuse of views, utilizing existing html partial files * reduction of calls per page click * separation of concerns - controllers separated to make for ease of readability, reduction of variable clashes * angular parameter use to control controller context * removal of redundant services
  Moved broke the user controller into two parts.  The admin related controllers are now separate.
  Adding navigation to the documentation page.
  Removing all unnecessary navigation partials and adding it to a central place.  It will be called only once during page loads.
  Removed the functionality of toggle and allowed for full modification of an application.
  I have secured the endpoints and added functionality to allows the angular application to automatically redirect to the login page if a http_code of 400/401 is received.
  reverted mongoose.connect arguments in db.js
  cirrected mislabelling in the swagger.yaml for the resources.
  Just ran through and auto indented a few other pages.
  Adding postinstall instructions for npm that allows bower to install the necessary dependencies.
  changed the tab spaces in the style.css to 4
  removed the comments from profile.html, removed the referencing to the navigation and the footer HTML from the profile.html, used angular for date conversion rather thaqn javascript parsing, and reverted the monngoose.connect arguments.
  modified the UserCtrl to allow the dev information to be retrievable from the scope so that the dev can see his/her information on the new profile page that was set up with a respective route
  Moving CSS to external file.
  Remove console logs and replaced the link to the ABIS website.
  ...

# Conflicts:
#	package.json
#	routes/resources/farmer/farmer.js

nickwxo commented:
This PR can now be reviewed again.

nickwxo commented:
Actually no - just remembered that the logging hasn't been implemented for the new (non-farmer endpoints using the Sequelize library). Will implement that and update again this evening.

paradoxxjm commented:
@nick22891 what's the status of the pull request?

matjames007 reviewed (approved):
No issues found during review.

matjames007 commented:
@nick22891 may need merge `develop` before I can proceed.

Commit:
Resolving merge conflicts on the branch. Merge branch 'develop' into harvest-additional-endpoints-impl

* develop: (23 commits)
  Update README.md
  Update README.md to include waffle badge
  Update db.js
  Update db.js
  The 'Farmers' resource is now the example of how all resources will be documented. This branch is still under development as the rest of resource documentation will be translated.
  testing out security definitions in docs along with correct get params
  Fixed table pagination.
  Search functionality for users and apps on the Admin Dashboard.
  Re-implementing searching on the app and user logs.
  Update db.js
  user.js and user-ctrl.js has been cleaned of incomplete code that was setup for the functionality of changing a user's password. This feature will be setup separately.
  I wasn't satisfied with the redundant code I had and I also discovered a bug that wouldn't check if the user was administrator in a particular use case.
  Using this commit to close out Issue #77 and apply a class based on the type of user login.
  Added CSS and other styling elements for the Admin Dashboard UI.
  UPDATE: profile.html
  UPDATE: profile.html
  added required field indication on signup.html and signin.html
  trying to change password
  Upon looking on other http status code conventions I have reverted to using error code 401 for all things authentication related.  400 is used for bad requests and as such I have built in a check that looks if the platform is sending a validation error from mongo and then set the codes to 400 along with the parameter(s) at fault.  This can be used to construct an error message from the frontend application.
  *BEFORE: After signing up, if "Sign in with a different account" is selected it goes to "/signin.html" which displays "{"message":"Not Found","error":{}}" *AFTER: Now redirects t0: "/#/signin" :thumbsup:
  ...

# Conflicts:
#	package.json

nickwxo commented:
@matjames007 Conflicts resolved.
